### PR TITLE
#793: cleanup XML parsing exception handling

### PR DESCRIPTION
**Pull request by lorban**

### CODE REVIEW

File: xml/src/main/java/org/ehcache/xml/ConfigurationParser.java [hunk 1]
```diff
@@ -115,24 +115,14 @@ public ConfigurationParser(String xml, URL... sources) throws IOException, SAXEx
     factory.setXIncludeAware(true);
     factory.setSchema(XSD_SCHEMA_FACTORY.newSchema(schemaSources.toArray(new Source[schemaSources.size()])));
 
-    final DocumentBuilder domBuilder;
-    try {
-      domBuilder = factory.newDocumentBuilder();
-    } catch (ParserConfigurationException e) {
-      throw new AssertionError(e);
-    }
+    DocumentBuilder domBuilder = factory.newDocumentBuilder();
```
Comment (chrisdennis) [inline xml/src/main/java/org/ehcache/xml/ConfigurationParser.java]: I would still vote that a ParserConfigurationException should be converted to an AssertionError here.

**Comment by chrisdennis** [comment]
Looks fine other than my vote on that ParserConfigurationException... another vote for your position would allow me to merge this.

File: xml/src/main/java/org/ehcache/xml/ConfigurationParser.java (hunk 1 above)
Comment (GaryWKeim) [inline xml/src/main/java/org/ehcache/xml/ConfigurationParser.java]: I thought assertions were for things that normally shouldn't occur. Messing up you XML is normal.

Comment (ljacomet) [inline xml/src/main/java/org/ehcache/xml/ConfigurationParser.java]: Changes looks good to me, not that I agree with Gary :wink:

**Pull request merged**